mlflow/pyfunc/model.py

```diff
@@ -0,0 +1,275 @@
+import os
+import tempfile
+import shutil
+import sys
+import yaml
+from abc import ABCMeta, abstractmethod
+
+import cloudpickle
+
+import mlflow.pyfunc
+import mlflow.utils
+from mlflow.exceptions import MlflowException
+from mlflow.models import Model
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE, RESOURCE_ALREADY_EXISTS
+from mlflow.tracking.utils import _download_artifact_from_uri
+from mlflow.utils.environment import _mlflow_conda_env
+from mlflow.utils.model_utils import _get_flavor_configuration
+from mlflow.utils.file_utils import TempDir, _copy_file_or_tree
+
+DEFAULT_CONDA_ENV = _mlflow_conda_env(
```

Comment:
This is only the default for serialized PythonModel objects. I guess this file is only concerned with those, although from a logical perspective, I would imagine the code to be shared almost identically with the data_path/loader_module pathway. For example, we could imagine just instantiating a default PythonModel which loads the provided loader_module -- I guess the main downside of that approach is that we would need to require cloudpickle for all pyfunc model environments, which may actually be pretty severe.

Anyway, all that to say, we should at least comment at this point in the code what this is the default for, in addition to having the module-level documentation saying that all concepts here are related only to PythonModel serialization (as opposed to loader_module saving of pyfunc models; unless I'm wrong about that distinction).

Reply:
I wonder if we should consider at some point shading cloudpickle into mlflow, to fix the version that we depend on and make it independent of the customer's environment.